File: src/rngGame/stats/Gear.java
package rngGame.stats;

public class Gear extends Item {

	private int hp;

	private double res;

	private double dgc;

	private int atk;

	public Gear(String ordner, String item, Rarity rarity) {
		super(ordner, item, rarity);
	}

	public int getHp() {
		return hp;
	}

	public double getRes() {
		return res;
	}

	public double getDgc() {
		return dgc;
	}

	public int getAtk() {
		return atk;
	}

	protected void setHp(int hp) {
		this.hp = hp;
	}

	protected void setRes(double res) {
		this.res = res;
	}

	protected void setDgc(double dgc) {
		this.dgc = dgc;
	}

	protected void setAtk(int atk) {
		this.atk = atk;
	}

	@Override
	public String toString() {
		return getClass().getSimpleName() + " [rarity=" + getRarity() + ", hp=" + hp + ", res=" + res + ", dgc=" + dgc + ", atk=" + atk + "]";
	}

}

File: src/rngGame/stats/Gear.kt
package rngGame.stats

open class Gear(ordner: String, item: String, rarity: Rarity): Item(ordner, item, rarity) {

	var hp = 0
		protected set

	var res = 0.0
		protected set

	var dgc = 0.0
		protected set

	var atk = 0
		protected set

	override fun toString(): String {
		return javaClass.getSimpleName() + " [rarity=" + rarity + ", hp=" + hp + ", res=" + res + ", dgc=" + dgc + ", atk=" + atk + "]"
	}

}
